CMakeLists.txt

```diff
@@ -115,7 +115,7 @@ set(SOURCE_FILES
         lib/rbtree.h
         lib/rbtree.c
         lib/memory.h
-        lib/memory.c
+        lib/memory.cpp
```

Comment:
Please look around and make your changes inline with existing implementation. The wrapper should be in cpp folder and added to source list of SOURCE_FILES_CPP_WRAPPER.
I would prefer to have a wrapper on malloc directly, in https://github.com/snowflakedb/libsnowflakeclient/blob/master/cpp/util/SnowflakeCommon.cpp
or if you want create a new file,
check and throw exception there, and call the C wrapper in memory.c (no need to change the existing file name)
In such case we don't need to check the return value in memory.c since it's been checked in the wrapper already.

Please also make the behavior configurable otherwise it would be a breaking change for libsnowflakeclient, changing the error handling from exit to throw exception. That's good for ODBC but might be an issue for any other application using libsfc that doesn't expect exception.